### PR TITLE
Employee Details Page

### PR DESCRIPTION
**Pull request by bibas07**
Employee Details Page done - TED001

### CODE REVIEW

File: src/helpers/BaseHelper/employeeCreation.helper.ts [hunk 1]
```diff
@@ -8,35 +8,42 @@ export class FillEmployeeCreationForm extends BaseHelper {
...
-            name: 'email',
-        });
-        await this.fillText(employeeInfo.employee_code, {
-            name: 'identifier',
-        });
-        await this.selectOption({
-            input: employeeInfo.department,
-            placeholder: 'Select  Department',
-        });
+        if (await employeeInfo.name)
```
Comment (sirjandrn25) [inline src/helpers/BaseHelper/employeeCreation.helper.ts]: you can use formHelper.fillFormInputInformation

File: src/helpers/BaseHelper/form.helper.ts [hunk 2]
```diff
@@ -86,13 +86,22 @@ export class FormHelper extends BaseHelper {
...
+     * @param inputName - The name of the input field to check.
+     * @returns A promise that resolves when the input field is confirmed to be empty.
+     */
     async isInputFieldEmpty(inputName: string): Promise<void> {
-        const name_field = await this.locate('input', {
+        const nameField: string = await this.locate('input', {
             name: inputName,
         })._locator.inputValue();
-        expect(name_field, chalk.red('Name field value')).toBe('');
+        expect(nameField).toBe('');
```
Comment (sirjandrn25) [inline src/helpers/BaseHelper/form.helper.ts]: why you remove chalk.red?

File: src/helpers/BaseHelper/form.helper.ts [hunk 3]
```diff
@@ -113,6 +122,34 @@ export class FormHelper extends BaseHelper {
...
+        }
+        if (!targetClick) return;
+        await this.click({
+            selector: 'textarea',
+            name: targetClick,
+        });
+    }
+
+    public async uploadDocument(fileName: ObjectDto): Promise<void> {
+        console.log(chalk.blue('Uploading Document ...'));
```
Comment (sirjandrn25) [inline src/helpers/BaseHelper/form.helper.ts]: you can use fileHelper for upload document

File: src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts [hunk 4]
```diff
@@ -210,3 +224,104 @@ export class AddEmployeeCreation extends BaseHelper {
...
 }
+
+export class EmployeeDetailsPage extends BaseHelper {
+    public breadCrumb: BreadCrumbHelper;
+    public listing: ListingHelper;
+    constructor(page: any) {
+        super(page);
+        this.listing = new ListingHelper(page);
+    }
+
```
Comment (sirjandrn25) [inline src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts]: in all public method you have to use public keyword becouse rumesh sir want to make this standard

File: src/helpers/BaseHelper/employeeCreation.helper.ts (hunk 1 above)
Comment (bibas07) [reply src/helpers/BaseHelper/employeeCreation.helper.ts]: done

File: src/helpers/BaseHelper/form.helper.ts (hunk 2 above)
Comment (bibas07) [reply src/helpers/BaseHelper/form.helper.ts]: this can be use for every input field

Comment (bibas07) [reply src/helpers/BaseHelper/form.helper.ts]: added again

File: src/helpers/BaseHelper/form.helper.ts (hunk 3 above)
Comment (bibas07) [reply src/helpers/BaseHelper/form.helper.ts]: done

File: src/helpers/BaseHelper/base.helper.ts [hunk 5]
```diff
@@ -811,20 +759,18 @@ export class BaseHelper {
...
-            const tds = await row.locator(
-                '//div[contains(@class,"table-cell")]'
-            );
+            const row = rows.nth(i);
+            const tds = row.locator('//div[contains(@class,"table-cell")]');
             for (let j = 0; j < (await tds.count()); j++) {
                 const cell = await tds.nth(j).innerText();
                 if (cell === name) {
-                    const Button = await tds.nth(cellno).locator(`${locator}`);
+                    const Button = tds.nth(cellno).locator(`${locator}`);
```
Comment (sirjandrn25) [inline src/helpers/BaseHelper/base.helper.ts:773]: why you remove await from this

Comment (bibas07) [reply src/helpers/BaseHelper/base.helper.ts:773]: it's not necessary there

File: src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts [hunk 6]
```diff
@@ -63,35 +97,35 @@ export class AddEmployeeCreation extends BaseHelper {
...
-            `//a[text()="${employeeCreationInfo.designation}"]`
+            `//a[text()="${employeeCreationInfo.designation_id}"]`
         );
     }
 
     async getEmployeeGrade() {
         const parentLocator = await this.employeeCodeLocator();
         return parentLocator.locator(
-            `//p[text()="${employeeCreationInfo.grade}"]`
+            `//p[text()="${employeeCreationInfo.grade_id}"]`
```
Comment (sirjandrn25) [inline src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts]: use listing helper for find rows

Comment (bibas07) [reply src/helpers/EmplyeeCreationHelper/employeeCreation.helper.ts]: done

**Pull request merged**